### ISSUE TITLE
Allow to disable creation of name index, how to implement?

### ISSUE THREAD

**Issue by Nakaner** [graphhopper/graphhopper#1608]
I am currently working on a [tool](https://github.com/geofabrik/osmi_routing) to find potential routing errors in OpenStreetMap – at the moment limited unconnected ways. While I already use my own GraphHopper fork with added callbacks to the `OSMReader` class, I would like to avoid more patches if the feature is accepted upstream. [1] I don't need a name index in my use case and would like to be able to disable it.

The constructor of the `NameIndex` class and its `init` method is called by the `BaseGraph` class. Adding another parameter to the constructor of `BaseGraph` would be the simplest solution but looks a bit hacky. Would you accept such a pull request or do you prefer an approach cleaning up how properties which can be controlled using the YAML configuration are handled in GraphHopper. The `GraphHopper` class is already full of various string and numeric properties which could be moved into a dedicated `GHSettings` class.

I am aware that an disabled name index needs to be handled in `RouteResource` and its tests.

[1] This does not mean that my desire to have callbacks in `OSMReader` was rejected. The pain of maintaining my own fork was not strong enough to discuss potential changes in detail and implement them etc. :-)

**Comment by karussell** [comment]
What would be the benefit of omitting the NameIndex?

**Comment by Nakaner** [comment]
In my use case, I iterate over all nodes in the graph and look up all nodes with one edge only in the location index in order to retrieve nearby nodes the could be connected to. I do some routing requests to filter out false positives but I am neither interested in navigation instructions nor index lookups based on names. Omitting the index saves time to generate it.

**Comment by karussell** [comment]
> Omitting the index saves time to generate it

Have you measured the impact? I would say the storage size is less than 5% (maybe even 1%) and the same for the time to create it.

**Comment by Nakaner** [comment]
I did an import for Germany with a flag encoder accepting all roads. It took a little bit more than 9 minutes from start until the end of the removal of the subnetworks. There wasn't any measurable difference if I comment out the name index. I will repeat this on are different machine with a larger dataset later.

In the meanwhile, I found out that there is an option `datareader.instructions` in the YAML configuration. If it is set to false, no entries will be inserted into the index which is sufficient for my use case.
